apps/api-extractor-model/src/aedoc/ReleaseTag.ts

```diff
@@ -67,8 +67,9 @@ export namespace ReleaseTag {
         return '@beta';
       case ReleaseTag.Public:
         return '@public';
+      default:
+        throw new Error('Unsupported release tag');
     }
-    throw new Error('Unsupported release tag');
```

Comment:
Was this change intentional?

Reply:
Yeah, the TS `noUnreachableCode` compiler option complains about these.

Reply:
🤔 It's off-topic for this PR.  But the TypeScript compiler is plainly wrong to complain about this code.  It seems like a reason to disable that compiler option in our rigs.

Lately the compiler's increasingly strict options have been writing checks that its type analysis cannot cash. It makes me wonder whether our config should be a little more lax. For `strictNullChecks` was awesome because it catches lots of bugs, whereas for `noImplicitAny` and `strictPropertyInitialization` the payoff seems dubious compared to the hassle.

Reply:
We've told the compiler that the only possible options for `ReleaseTag` are the ones that are explicitly covered by the `case`s, so I'd argue that the compiler is actually "right" here.

Reply:
`strictPropertyInitialization` is basically an offshoot of `strictNullChecks`, but for class properties.